File: src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/Grouping.java
package de.pbauerochse.worklogviewer.youtrack.v20174;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import de.pbauerochse.worklogviewer.youtrack.domain.GroupByCategory;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Grouping {

    private final GroupByCategory groupByCategory;

    public Grouping(GroupByCategory groupByCategory) {
        this.groupByCategory = groupByCategory;
    }

    public String getId() {
        return groupByCategory.getId();
    }

    @JsonProperty("$type")
    public String getTypeHeader() {
        return "jetbrains.charisma.smartui.report.time.WorkItemBasedGrouping";
    }
}

File: connectors/v20182/src/main/java/de/pbauerochse/worklogviewer/connector/v2018/domain/report/Grouping.kt
package de.pbauerochse.worklogviewer.connector.v2018.domain.report

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import de.pbauerochse.worklogviewer.connector.GroupByParameter

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "\$type")
@JsonSubTypes(
    JsonSubTypes.Type(value = WorkItemBasedGrouping::class, name = "jetbrains.youtrack.reports.impl.time.gap.WorkItemBasedGrouping"),
    JsonSubTypes.Type(value = FieldBasedGrouping::class, name = "jetbrains.youtrack.reports.impl.time.gap.FieldBasedGrouping")
)
interface Grouping : GroupByParameter